lib/restforce/concerns/streaming.rb

```diff
@@ -30,6 +31,65 @@ def faye
           client.bind 'transport:up' do
             Restforce.log "[COMETD UP]"
           end
+
+          client.add_extension ReplayExtension.new(replay_handlers)
+        end
+      end
+
+      def replay_handlers
+        @_replay_handlers ||= {}
+      end
+
+      class ReplayExtension
+        def initialize(replay_handlers)
+          @replay_handlers = replay_handlers
+        end
+
+        def incoming(message, callback)
+          channel = message.fetch('channel').gsub('/topic/', '')
+          replay_id = message.fetch('data', {}).fetch('event', {})['replayId']
```

Comment:
Seems to me that saving the replay_id should occur after the callback successfully return